### PR TITLE
test: AppVeyor Jobs Running Against Nightly Binaries

### PR DESCRIPTION
**Pull request by mildaniel**
#### Which issue(s) does this change fix?

N/A

#### Why is this change necessary?
To run integration tests against production binaries.

#### How does it address the issue?
Creates new AppVeyor configurations that download the nightly binaries on Windows and Linux and then run the integration test suite against them.

#### What side effects does this change have?
N/A

#### Mandatory Checklist
**PRs will only be reviewed after checklist is complete**

- [x] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document if needed ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [] Write/update unit tests
- [x] Write/update integration tests
- [ ] Write/update functional tests if needed
- [x] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: appveyor-linux-binary.yml [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+  # Install AWS CLI
+  - sh: "virtualenv aws_cli"
+  - sh: "./aws_cli/bin/python -m pip install awscli"
+  - sh: "PATH=$(echo $PWD'/aws_cli/bin'):$PATH"
+
+  # Install pytest
+  - sh: "pip3 install -r requirements/pre-dev.txt"
+  - sh: "pip3 install -r requirements/dev.txt"
+  - sh: "pip3 install -r requirements/base.txt"
+  - sh: "pytest --version"
```
Comment (moelasmar) [inline appveyor-linux-binary.yml:77]: I think it is better if we installed these libraries in a virtual environment to make sure that we isolate our test cases from any libraries that got installed here. I also think if just install the libraries that are required for the integration testing only.

**Review by moelasmar — APPROVED** [review]

File: appveyor-linux-binary.yml (hunk 1 above)
Comment (mildaniel) [reply appveyor-linux-binary.yml]: I changed it to download the dependencies into a virtual env, but we require a bunch of these dependencies to run the integration tests. Might not be worth trying to narrow it down right now.

File: appveyor-linux-binary.yml [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+    if [[ -n $BY_CANARY ]];
+      then echo Logging in Public ECR; aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws;
+    fi"
+
+
+build_script:
+  - curl -L https://github.com/aws/aws-sam-cli/releases/download/sam-cli-nightly/aws-sam-cli-linux-x86_64.zip -o aws-sam-cli-linux-x86_64.zip
+  - unzip aws-sam-cli-linux-x86_64.zip -d sam-installation
+  - sudo ./sam-installation/install
+  - sudo mv /usr/local/bin/sam-nightly /usr/local/bin/sam
```
Comment (mndeveci) [inline appveyor-linux-binary.yml:155]: nit: worth adding some comment that we are overriding executable name (`sam-nightly`) with `sam` here.

File: appveyor-linux-binary.yml [hunk 3]
```diff
@@ -0,0 +1,296 @@
+version: 1.0.{build}
```
Comment (mndeveci) [inline appveyor-linux-binary.yml:1]: There are too many `appveyor*` files at the root level, we might want to move them under `.appveyor` folder.

And there is some level of duplication between these files, I wonder if we can use some sort of "import" and move common configuration into separate files.

**Review by mndeveci — APPROVED** [review]
Thanks for working on this, left some unblocking comments.

**Pull request merged**